### PR TITLE
feat: Add new event - admin user

### PR DESCRIPTION
**Pull request by vivianamck**
## Issue ticket number and link

https://trello.com/c/a6lhWk9L

## Checklist before requesting a review
- [X] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics?
- [ ] Will this be part of a product update? If yes, please write one phrase about this update.

## Pull request type

Please check the type of change your PR introduces:

- [X] :sparkles: Feature
- [ ] :bug: Bugfix
- [ ] :pencil2: Fixing typos
- [ ] :lipstick: :art: Code style update (formatting, renaming)
- [ ] :recycle: Refactoring (no functional changes, no api changes)
- [ ] :zap: Performance
- [ ] :pencil: Documentation content changes
- [ ] :white_check_mark: Unit testing
- [ ] :ok_hand: Updating due to code review suggest
- [ ] :rocket: Release: v
- [ ] :wrench: Add or update configuration files
- [ ] :hammer: Add or update development scripts
- [ ] :bricks: Infrastructure related changes

## What is the current module?

- Page add non functional event

## What is the new behavior?

- Added functionality add event
- Updated  add event Endpoint 
- Added protected route to add event
- Update test to event type

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: src/api/events.js [hunk 1]
```diff
@@ -37,8 +37,9 @@ export default class Events extends baseRequest {
     return data.data
   }
 
-  add = (event) => {
-    return this.post(this.method, event)
+  add = async (event) => {
+    const response = await this.post(this.methodAll, event)
+    return response
   }
```
Comment (jlora-mck) [inline src/api/events.js:43]: I think we can do this, but if you wanna keep this function like you have it right now is also fine
```suggestion
  add = async (event) => ( await this.post(this.methodAll, event) )
```

File: src/components/DropDown/SelectWithLoading.tsx [hunk 2]
```diff
@@ -34,8 +34,8 @@ export interface SelectWithLoadingProps {
   attributeOptions: ElementOption[]
   error: boolean
   errorMessage: string
-  onChange: () => void
-  onBlur: () => void
+  onChange: (event:any) => void
```
Comment (jlora-mck) [inline src/components/DropDown/SelectWithLoading.tsx:37]: Im sure we can type this event, please change the `any`

File: src/components/EventView/EventView.spec.tsx [hunk 3]
```diff
@@ -1,40 +1,94 @@
...
+    expect(headquarter).toBeInTheDocument()
+    expect(type).toBeInTheDocument()
+    expect(tags).toBeInTheDocument()
+    expect(description).toBeInTheDocument()
+    expect(phoneNumber).toBeInTheDocument()
     expect(salesEventType).toBeInTheDocument()
+    expect(mockOnChange).resolves
+
+    expect(true).toBe(true)
+
```
Comment (jlora-mck) [inline src/components/EventView/EventView.spec.tsx:67]: please remove space

File: src/components/EventView/EventView.spec.tsx [hunk 4]
```diff
@@ -1,40 +1,94 @@
...
     expect(eventDate).toBeInTheDocument()
     expect(address).toBeInTheDocument()
+    expect(headquarter).toBeInTheDocument()
+    expect(type).toBeInTheDocument()
+    expect(tags).toBeInTheDocument()
+    expect(description).toBeInTheDocument()
+    expect(phoneNumber).toBeInTheDocument()
     expect(salesEventType).toBeInTheDocument()
+    expect(mockOnChange).resolves
+
```
Comment (jlora-mck) [inline src/components/EventView/EventView.spec.tsx:65]: please remove spaces

File: src/components/EventView/EventView.spec.tsx [hunk 5]
```diff
@@ -1,40 +1,94 @@
...
+      "type": "Sales",
+      "description": "Dress up green!",
+      "tags": "Devops",
+      "phoneNumber": "" 
+  };
+
+    axios.post = jest.fn();
+    const mockedAxios = axios as jest.Mocked<typeof axios>;
+
+    mockedAxios.post.mockResolvedValueOnce(service);
```
Comment (jlora-mck) [inline src/components/EventView/EventView.spec.tsx:86]: All the mocking should take place outside the describe method

File: src/components/EventView/EventView.tsx [hunk 6]
```diff
@@ -40,50 +45,98 @@ export interface EventViewProps {
...
+    },
+    {
+      "id": "Design",
+      "name": "Design"
+    },
+    {
+      "id": "Devops",
+      "name": "Devops"
+    }
+  ]
```
Comment (jlora-mck) [inline src/components/EventView/EventView.tsx:85]: I understand that these are mocks, so lets create a mock file and import it here please

File: src/components/EventView/EventView.tsx [hunk 7]
```diff
@@ -40,50 +45,98 @@ export interface EventViewProps {
...
   const classes = useStyles()
+  const history = useHistory()
+
+  const [valEventName, setEventName] = useState(eventName)
+  const [selectedEventType, setSelectedEventType] = useState(eventType)
+  const [valEventDescription, setEventDescription] = useState(eventDescription)
+  const [valEventDate, setEventDate] = useState(eventDate)
+  const [valAddress, setAddress] = useState(address)
+  const [valPhoneNumber, setPhoneNumber] = useState(phoneNumber)
+  const [valEventTag, setEventTag] = useState(eventTag)
```
Comment (jlora-mck) [inline src/components/EventView/EventView.tsx:100]: val stands for value ? i think we can remove that, code will look more legible without it

Comment (jlora-mck) [reply src/components/EventView/EventView.tsx:100]: All these useState should be typed

File: src/components/EventView/EventView.tsx [hunk 8]
```diff
@@ -40,50 +45,98 @@ export interface EventViewProps {
...
   }
 
-  const handleRequiredFieldBlurred = () => {}
-
-  const handleCancelButton = () => {}
-
-  const handleSubmitButton = () => {}
-
   return (
     <Grid container className={classes.container}>
```
Comment (jlora-mck) [inline src/components/EventView/EventView.tsx:141]: if this is a form why we dont have a `<form>` tag ?

File: src/api/events.js (hunk 1 above)
Comment (vivianamck) [reply src/api/events.js]: Done

File: src/components/EventView/EventView.spec.tsx (hunk 3 above)
Comment (vivianamck) [reply src/components/EventView/EventView.spec.tsx]: Done

File: src/components/EventView/EventView.spec.tsx (hunk 4 above)
Comment (vivianamck) [reply src/components/EventView/EventView.spec.tsx]: Done

File: src/components/EventView/EventView.tsx (hunk 6 above)
Comment (vivianamck) [reply src/components/EventView/EventView.tsx]: Done

File: src/components/EventView/EventView.spec.tsx (hunk 5 above)
Comment (vivianamck) [reply src/components/EventView/EventView.spec.tsx]: Done

File: src/components/EventView/EventView.tsx (hunk 7 above)
Comment (vivianamck) [reply src/components/EventView/EventView.tsx]: Done

File: src/components/DropDown/SelectWithLoading.tsx (hunk 2 above)
Comment (vivianamck) [reply src/components/DropDown/SelectWithLoading.tsx]: Done

File: src/components/DropDown/SelectWithLoading.tsx [hunk 9]
```diff
@@ -7,6 +7,7 @@ import {
   makeStyles,
   createStyles,
 } from '@material-ui/core'
+import { ChangeEvent } from 'react'
```
Comment (jlora-mck) [inline src/components/DropDown/SelectWithLoading.tsx:10]: React impor its always the first line of the file

File: src/mocks/headquarter.js
Comment (jlora-mck) [inline src/mocks/headquarter.js:1]: space at the end

File: src/components/DropDown/SelectWithLoading.tsx (hunk 9 above)
Comment (vivianamck) [reply src/components/DropDown/SelectWithLoading.tsx]: Done

File: src/mocks/headquarter.js
Comment (vivianamck) [reply src/mocks/headquarter.js:1]: Done

**Review by jlora-mck — APPROVED** [review]

File: src/mocks/tags.js
[the remark this answers is not in the document]
Comment (vivianamck) [reply src/mocks/tags.js:1]: Done

**Review by Sofiag8 — APPROVED** [review]

**Review by FelipeLosada1996 — APPROVED** [review]

**Review by jose-sal-y-rosas — APPROVED** [review]
👍🏼

**Pull request merged**